src/sinks/datadog/logs.rs

```diff
@@ -244,14 +249,24 @@ impl HttpSink for DatadogLogsJsonService {
 
         self.config.encoding.apply_rules(&mut event);
 
+        let api_key = event
+            .metadata_mut()
+            .datadog_api_key
+            .take()
+            .unwrap_or_else(|| self.config.api_key.clone());
+
         let (fields, metadata) = event.into_log().into_parts();
+        let json_event = json!(fields);
+
         Some(EncodedEvent {
-            item: json!(fields),
+            item: PartitionInnerBuffer::new(json_event, api_key),
```

Comment:
Is `PartitionInnerBuffer` appropriate here? I'm confused about the logic. I got worried about the logic to pull `api_key`, specifically, what happens if two `Event` instances pass through `encode_event` with different API keys do they pollute the buffer? Then I realized we aren't actually batching into `PartitionInnerBuffer` here unless I'm very mistaken. Are we using `PartitionInnerBuffer` to associate a single `Event` with an API key, in case the event doesn't have a key? 

This PR does introduce `PartitionInnerBuffer` but doesn't seem to make use of the batching inherent in that type. If I'm right and not missing some broader context, why not mutate the `Event` to include a key if the key is missing, or adjust input and output to be a tuple?

Reply:
It does look like `build_request` in `DatadogLogsTextService` is written with the assumption that there are multiple events in the `PartitionInnerBuffer` so I'm probably only half understanding something here.

Reply:
Ah, hmm, that was confusing. Thank you for the clarification.